packages/docs-site/src/content/docs/docs/plugins/overview.mdx

```diff
@@ -12,6 +12,96 @@ Our first set of plugins are designed to help you manage secrets:
+
+- Sensitive values that are needed to access the plugin's service
+- Non-sensitive values that are needed to configure the plugin
+
+Because plugins are typically used to access sensitive items themselves, this introduces a slight chicken-and-egg problem. In this case, DMNO allows you to define the plugin inputs in the `schema` but not provide a value. The sensitive value is then set via an _override_. 
+
+:::tip[More on overrides]
+For more on how overrides work and how to use them, see the [Setting overrides via `.env` files](https://dmno.dev/docs/guides/env-files/#setting-overrides-via-env-files).
+:::
+
+### Wiring up the inputs
+
+Because DMNO provides plugin-specific types, we can automatically inject the correct type for the plugin input. This means you don't need to manually wire up the inputs when you initialize the plugin. Our config engine will do this for you. 
+
+For example: 
+
+```ts
+import { SomePlugin, SomePluginTypes } from '@dmno/some-plugin';
+
+// by default, access token will be injected using types
+const somePluginInstance = new SomePlugin('some-plugin');
+
+// or you can explicitly wire it up by path
+const somePluginInstance2 = new SomePlugin('some-plugin', {
+  someToken: configPath('..', 'SOME_TOKEN'),
+  someOtherToken: configPath('..', 'SOME_OTHER_TOKEN'),
+});
+
+export default defineDmnoService({
+  schema: {
+    SOME_TOKEN: {
+      // this type allows us to inject the correct value from the config item
+      extends: SomePluginTypes.someToken,
+    },
+    SOME_OTHER_TOKEN: {
+      extends: SomePluginTypes.someOtherToken,
+    },
+  },
+});
+```
+
+:::caution[Be careful]
+Because of the nature of type-based injection, you can only do this for a single config item of a given type. If you have multiple config items of the same type, like two of the same type of token for two different plugin instances, you'll need to explicitly wire them up.
+:::
+
+
+
+### Multiple plugin instances
+
+Plugins support multiple instances allowing you to compose your configuration in a flexible way. For example, with the 1Password plugin you could create separate instances for your production and non-production vaults. Each instance can have its own settings, allowing you to manage per-environment or per-service secrets.
+
+When defining a plugin instance, you provide an `id` which will be used to identify the instance in various places, such as the CLI, UI, and when injecting it into other services.
+
+### Injecting plugin instances in monorepo services
+
+In a monorepo, you might want to use the same plugin instance across multiple services. If you will be using the same settings for each service, you can initialize a plugin instance once in a parent service as seen above, and then inject it in child services. Note that the injected plugin instance must use the same id we set during initialization.
```

Comment:
you could mention that it is injects the pre-configured instance, avoiding the need to pull in the config and wire it up in each service